### ISSUE TITLE
Low OK-VQA accuracy (~26.8 %) when reproducing the “no-RL” setting

### ISSUE THREAD

**Issue by Solus-sano** [ControlNet/HYDRA#10]
Hi team,

I’m trying to replicate the no-RL OK-VQA ACC reported in the paper (46.9 % in the ablation, and 48.6 % full model).
Following the README I launched one executor and one inference worker:

```Bash
# 1. start the executor
python -m hydra_vl4ai.executor \
        --base_config ./config/okvqa.yaml \
        --model_config ./config/model_config_1gpu.yaml

# 2. run inference
python main.py \
        --data_root  ../datasets/OKVQA \
        --base_config config/okvqa.yaml \
        --model_config config/model_config_1gpu.yaml
```
After ~3 000 examples I evaluated the partial file:
```Bash
python evaluate.py result/result_okvqa.jsonl okvqa
```
and obtained 26.8 % accuracy – far below the 46 – 48 % reported. Are there any common pitfall that could explain a 20 pp gap? (e.g. wrong split, val vs. test, missing caption model or third-party GPT-4o-mini proxy be the cause?)

Environment
- CUDA 11.8, torch 2.1.2, Python 3.11.13
- 1× RTX3090 (24 GB)
- OpenAI model: gpt-4o-mini (from third-party proxy)
- Datasets follow the expected layout (COCO train/val2014 + OK-VQA JSON).

My inference result can be downloaded [here](https://drive.google.com/file/d/1IRUHx5oeALq_n7WgU7PaeVOjtFNc-L2G/view?usp=sharing)

Thanks for the great work—any guidance that helps me align with the reported baseline would be much appreciated!

**Comment by pokerme7777** [comment]
Hi there,

Thanks for your attention.

I reviewed your inference results. There is a high proportion of "null" and "continue" outputs, which suggests that the model couldn't generate a valid answer within x turns. (x is the hyperparameter you've set.)

If you check the supplementary material of [VisProg](https://arxiv.org/pdf/2211.11559) and the[ ViperGPT](https://github.com/cvlab-columbia/viper/issues/24) code, you’ll see similar issues caused by incorrect GPT outputs. This also indicates there might be some formatting issues in the outputs, which could be limiting the overall system performance.

I recommend slightly modifying the prompt to better align with the current checkpoint of GPT-4o-mini. Previously, we used GPT-3.5, and the same prompt worked well with that model. However, when switching checkpoints or models, it's often necessary to adapt the prompt accordingly.

I hope this helps you better understand or navigate our code.

**Comment by Solus-sano** [comment]
> Hi there,
> 
> Thanks for your attention.
> 
> I reviewed your inference results. There is a high proportion of "null" and "continue" outputs, which suggests that the model couldn't generate a valid answer within x turns. (x is the hyperparameter you've set.)
> 
> If you check the supplementary material of [VisProg](https://arxiv.org/pdf/2211.11559) and the[ ViperGPT](https://github.com/cvlab-columbia/viper/issues/24) code, you’ll see similar issues caused by incorrect GPT outputs. This also indicates there might be some formatting issues in the outputs, which could be limiting the overall system performance.
> 
> I recommend slightly modifying the prompt to better align with the current checkpoint of GPT-4o-mini. Previously, we used GPT-3.5, and the same prompt worked well with that model. However, when switching checkpoints or models, it's often necessary to adapt the prompt accordingly.
> 
> I hope this helps you better understand or navigate our code.

Does it mean that the agent system is very sensitive to prompts?
If I want to run it with llama 3.1 or llama 3.3, do I also need to customize the prompts for them?

**Comment by Solus-sano** [comment]
Thanks a lot for the quick-and-clear guidance! Really appreciate the help (and the awesome codebase) 😊.
